vulnerabilities/importers/github.py

```diff
@@ -0,0 +1,342 @@
+                    }
+
+                    unaffected_purls = {
+                        PackageURL(
+                            name=pkg_name, namespace=ns, version=version, type=pkg_type
+                        )
+                        for version in unaff_vers
+                    }
+
+                    cve_ids = set()
+                    ref_ids = set()
+                    vuln_desc = adv['node']['advisory']['summary']
+
+                    for vuln in adv['node']['advisory']['identifiers']:
+                        if vuln['type'] == 'CVE':
+                            cve_ids.add(vuln['value'])
+                        else:
+                            ref_ids.add(vuln['value'])
+                    for cve_id in cve_ids:
+                        adv_list.append(
+                            Advisory(
+                                cve_id=cve_id,
+                                summary=vuln_desc,
+                                impacted_package_urls=affected_purls,
+                                resolved_package_urls=unaffected_purls,
+                                reference_ids=ref_ids,
+                            )
+                        )
+        return adv_list
+
+    @staticmethod
+    def categorize_versions(
+        version_range: str, all_versions: Set[str]
+    ) -> Tuple[Set[str], Set[str]]:
+        version_range = RangeSpecifier(version_range)
+        affected_versions = {
+            version for version in all_versions if version in version_range
+        }
+        return (affected_versions, all_versions - affected_versions)
+
+
+class MavenVersionAPI:
+    def __init__(self):
+        self.cache = {}
+
+    def get(self, pkg_name: str) -> Set[str]:
+        return self.cache.get(pkg_name, set())
+
+    def load_to_api(self, pkg_name: str) -> None:
+
+        if pkg_name in self.cache:
+            return
+
+        artifact_comps = pkg_name.split(':')
+        endpoint = self.artifact_url(artifact_comps)
+        resp = requests.get(endpoint).content
+
+        try:
+
+            xml_resp = ET.ElementTree(ET.fromstring(resp.decode('utf-8')))
+            self.cache[pkg_name] = self.extract_versions(xml_resp)
+
+        except ET.ParseError:
+            self.cache[pkg_name] = set()
+
+    @staticmethod
+    def artifact_url(artifact_comps: List[str]) -> str:
+
+        base_url = 'https://repo.maven.apache.org/maven2/{}'
+        group_id, artifact_id = artifact_comps
+        group_url = group_id.replace('.', '/')
+        suffix = group_url + '/' + artifact_id + '/' + 'maven-metadata.xml'
+        endpoint = base_url.format(suffix)
+
+        return endpoint
+
+    @staticmethod
+    def extract_versions(xml_response: ET.ElementTree) -> Set[str]:
+
+        all_versions = set()
+        for child in xml_response.getroot().iter():
+            if child.tag == 'version':
+                all_versions.add(child.text)
+
+        return all_versions
+
+
+class NugetVersionAPI:
```

Comment:
Is there some code share between the various "VersionAPI" classes? What about moving the common parts in a super class? or some helper?

Reply:
That will be done in another separate PR, because this is not just limited to  this importer, rather almost every other importer in `vulnerabilities/importers` has one such `VersionAPI` class thingie.

Reply:
ok